pkg/debug/stresschaos/stresschaos.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stresschaos
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"code.cloudfoundry.org/bytefmt"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	cm "github.com/chaos-mesh/chaos-mesh/pkg/debug/common"
+)
+
+func Debug(ctx context.Context, chaos runtime.Object, c *cm.ClientSet) error {
+	stressChaos, ok := chaos.(*v1alpha1.StressChaos)
+	if !ok {
+		return fmt.Errorf("chaos is not stresschaos")
+	}
+	chaosStatus := stressChaos.Status.ChaosStatus
+	chaosSelector := stressChaos.Spec.GetSelector()
+
+	pods, daemons, err := cm.GetPods(ctx, chaosStatus, chaosSelector, c.CtrlCli)
+	if err != nil {
+		return err
+	}
+
+	for i := range pods {
+		podName := pods[i].GetObjectMeta().GetName()
+		cm.Print("[Pod]: "+podName, 0, cm.ColorBlue)
+		err := debugEachPod(ctx, pods[i], daemons[i], stressChaos, c)
+		if err != nil {
+			return fmt.Errorf("for %s: %s", podName, err.Error())
+		}
+	}
+	return nil
+}
+
+func debugEachPod(ctx context.Context, pod v1.Pod, daemon v1.Pod, chaos *v1alpha1.StressChaos, c *cm.ClientSet) error {
+	podName := pod.GetObjectMeta().GetName()
+	podNamespace := pod.GetObjectMeta().GetNamespace()
+	daemonName := daemon.GetObjectMeta().GetName()
+	daemonNamespace := daemon.GetObjectMeta().GetNamespace()
+
+	// cpu or memory chaos
+	isCPU := true
+	if cpuSpec := chaos.Spec.Stressors.CPUStressor; cpuSpec == nil {
+		isCPU = false
+	}
+
+	// get process path
+	cmd := fmt.Sprintf("cat /proc/cgroups")
+	out, err := cm.Exec(podName, podNamespace, cmd, c.KubeCli)
+	if err != nil {
+		return fmt.Errorf("run command '%s' failed with: %s", cmd, err.Error())
+	}
+	var cpuMountType string
+	if regexp.MustCompile("(cpu,cpuacct)").MatchString(string(out)) {
+		cpuMountType = "cpu,cpuacct"
+	} else {
+		cpuMountType = "cpu"
+	}
+
+	cmd = fmt.Sprintf("ps")
+	out, err = cm.Exec(podName, podNamespace, cmd, c.KubeCli)
```

Comment:
When running a command in target pod, keep in mind that the target container could be really simple, which means there is no `cat`, no `ps`...

e.g. The widely used `debian:slim` base image doesn't have `ps` command (which is provided by `procps` package)

Reply:
This could be a really difficult problem (especially for file related message collecting). For `ps` or other command which doesn't need to enter the mount namespace, you could run `nsenter` in `chaos-daemon` as an alternative.

However, for commands which need to enter the mount namespace (like `ls` and `cat`), things get much more complicated. A possible solution is to write a `c` or `rs` program to `setns` dynamically and read the folder or file.

Reply:
Thanks for the remind! I'll look into it

Reply:
just a reminder: `nsenter` also support running with `-m` option to enter the target process mount ns, and I suggest make an individual `nsenter` pkg, just like [kubectl-debug did](https://github.com/aylei/kubectl-debug/blob/1ab75d9b50c38a797c8634de01d48930e8147a23/pkg/nsenter/nsenter.go).

Reply:
@Colstuwjx Great to know that, thanks!

Reply:
@Colstuwjx `pkg/bpm` package has included the function to generate `nsenter` arguments, and it's widely used in `chaos-daemon` package :smile_cat: 

Although `nsenter` support changing mount namespace, `ls` and `cat` may not exist in the target mnt namespace, which is the problem :disappointed:

Reply:
Hi @YangKeao I think I fix this problem